### PR TITLE
[VOID-21][add]:prisma module

### PR DESCRIPTION
**Pull request by MoutasemHabboub**
# Description

`Small Description of the task/bug/feature`

## Checklist

- [ ] Covered Acceptance criteria
- [ ] Added unit tests
- [ ] Follows technical architecture or UI/UX designs

## Links, Screenshots or other media

`Screenshots or other media`

### CODE REVIEW

File: src/app.module.ts [hunk 1]
```diff
@@ -4,10 +4,10 @@ import { AppDatabaseModule } from '@app/app-database/app-database.module';
 import { AppConfigModule } from '@app/app-config/app-config.module';
 import { SharedModule } from '@app/modules/shared.module';
 import { AppDefaultController } from '@app/app-controllers/app-default.controller';
+import { AppPrismaModule } from './app-prisma/app-prisma.module';
```
Comment (falyoun) [inline src/app.module.ts]: Replace `./app-prisma` with `@app/app-prisma...`

File: src/app-prisma/prisma/schema.prisma [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+generator client {
+  provider = "prisma-client-js"
+}
+
+datasource db {
+  provider = "postgresql"
+  url      = env("DATABASE_URL")
+}
+model User {
+  id    Int     @id @default(autoincrement())
```
Comment (falyoun) [inline src/app-prisma/prisma/schema.prisma]: Id to be UUID4

File: src/app-prisma/prisma/schema.prisma [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+  email String  @unique
```
Comment (falyoun) [inline src/app-prisma/prisma/schema.prisma]: email to be of type `CITEXT`

File: src/app-prisma/prisma/schema.prisma [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+  name  String?
```
Comment (falyoun) [inline src/app-prisma/prisma/schema.prisma]: two columns `firstName`, `lastName`
one virtual column `name`

File: src/app-prisma/prisma/schema.prisma [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+  number  String?
```
Comment (falyoun) [inline src/app-prisma/prisma/schema.prisma]: What does `number` mean?, kindly use `phoneNumber` instead

File: .env [hunk 6]
```diff
@@ -0,0 +1,10 @@
+
```
Comment (falyoun) [inline .env]: `.env` files should not be pushed to Github, you can create `.env.example` with variables' names and without values

File: src/app-prisma/prisma/schema.prisma [hunk 7]
```diff
@@ -0,0 +1,18 @@
+// This is your Prisma schema file,
+// learn more about it in the docs: https://pris.ly/d/prisma-schema
+
+generator client {
+  provider = "prisma-client-js"
+}
+
+datasource db {
+  provider = "postgresql"
+  url      = env("DATABASE_URL")
```
Comment (falyoun) [inline src/app-prisma/prisma/schema.prisma]: I'd prefer to use variables, dbname, dbhost, etc..., as a result of that I can override the dbname from command line without the need to write the full url

Comment (falyoun) [reply src/app-prisma/prisma/schema.prisma]: As well as, config variables must be loaded from .yaml files, in other words, using configService

File: src/app-prisma/prisma.service.ts [hunk 8]
```diff
@@ -0,0 +1,15 @@
+import { INestApplication, Injectable, OnModuleInit } from '@nestjs/common';
+import { PrismaClient } from '@prisma/client';
+
+@Injectable()
+export class PrismaService extends PrismaClient implements OnModuleInit {
```
Comment (falyoun) [inline src/app-prisma/prisma.service.ts:7]: For example

```
constructor(private configService: ConfigService<EnvironmentVariables>) {
    /**
     * Get the database url from environmental variables and pass it in.
     */
    super({
      datasources: {
        db: {
          // url: configService.get('DATABASE_URL'),
          name: configService.get('DB_NAME')
        },
      },
    });
  }
```

File: src/app-prisma/prisma/schema.prisma (hunk 7 above)
Comment (falyoun) [reply src/app-prisma/prisma/schema.prisma]: Left the usage at the constructor of `PrismaService`

File: .env [hunk 9]
```diff
@@ -0,0 +1,10 @@
...
+
+# This was inserted by `prisma init`:
+# Environment variables declared in this file are automatically made available to Prisma.
+# See the documentation for more detail: https://pris.ly/d/prisma-schema#accessing-environment-variables-from-the-schema
+
+# Prisma supports the native connection string format for PostgreSQL, MySQL, SQLite, SQL Server, MongoDB and CockroachDB.
+# See the documentation for all the connection string options: https://pris.ly/d/connection-strings
+
+DATABASE_URL="postgresql://postgres:password@localhost:5432/mydb?schema=public"
```
Comment (falyoun) [inline .env]: Kindly keep in mind, we use `.yaml` files and not `.env`

File: src/app-prisma/prisma/schema.prisma (hunk 2 above)
Comment (MoutasemHabboub) [reply src/app-prisma/prisma/schema.prisma]: this related to task number [36](https://2024-roadmap.atlassian.net/browse/VOID-36?atlOrigin=eyJpIjoiNmU3MjM5ODY3YjAxNGEwMGI3YTczMTBjNWVlY2JmMDkiLCJwIjoiaiJ9) in which will be handled

File: src/app-prisma/prisma.service.ts (hunk 8 above)
Comment (MoutasemHabboub) [reply src/app-prisma/prisma.service.ts:5]: prisma service has no interaction with prisma schema.

Comment (falyoun) [reply src/app-prisma/prisma.service.ts:7]: @MoutasemHabboub https://dev.to/harryhorton/how-to-use-prisma-with-nestjs-3l1b

**Review by falyoun — APPROVED** [review]

**Pull request merged**